<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="GinoBonazza (ginoandrea.bonazza@usz.ch)" />


<title>Differential expression analysis in all cell types using mPAP</title>

<script src="site_libs/header-attrs-2.26/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/jqueryui-1.13.2/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>
<link href="site_libs/font-awesome-6.4.2/css/all.min.css" rel="stylesheet" />
<link href="site_libs/font-awesome-6.4.2/css/v4-shims.min.css" rel="stylesheet" />

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/main/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>



<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>









<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
details > summary > p:only-child {
  display: inline;
}
pre code {
  padding: 0;
}
</style>


<style type="text/css">
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #adb5bd;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script type="text/javascript">
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark the anchor link active (and if it's in a dropdown, also mark that active)
  var dropdown = menuAnchor.closest('li.dropdown');
  if (window.bootstrap) { // Bootstrap 4+
    menuAnchor.addClass('active');
    dropdown.find('> .dropdown-toggle').addClass('active');
  } else { // Bootstrap 3
    menuAnchor.parent().addClass('active');
    dropdown.addClass('active');
  }

  // Navbar adjustments
  var navHeight = $(".navbar").first().height() + 15;
  var style = document.createElement('style');
  var pt = "padding-top: " + navHeight + "px; ";
  var mt = "margin-top: -" + navHeight + "px; ";
  var css = "";
  // offset scroll position for anchor links (for fixed navbar)
  for (var i = 1; i <= 6; i++) {
    css += ".section h" + i + "{ " + pt + mt + "}\n";
  }
  style.innerHTML = "body {" + pt + "padding-bottom: 40px; }\n" + css;
  document.head.appendChild(style);
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before, .tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "\e259";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "\e258";
  font-family: 'Glyphicons Halflings';
  border: none;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-bs-toggle="collapse" data-target="#navbar" data-bs-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">DCM_snRNAseq</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li class="dropdown">
  <a href="#" class="dropdown-toggle" data-toggle="dropdown" role="button" data-bs-toggle="dropdown" aria-expanded="false">
    <span class="fa fa-navicon"></span>
     
    Analysis
     
    <span class="caret"></span>
  </a>
  <ul class="dropdown-menu" role="menu">
    <li>
      <a href="QC_integration_annotation.html">QC, integration, annotation</a>
    </li>
    <li>
      <a href="Cardiomyocytes_subclustering.html">Cardiomyocytes subclustering</a>
    </li>
    <li>
      <a href="Cardiomyocytes_DA_DE.html">Cardiomyocytes differential expression</a>
    </li>
  </ul>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        <li>
  <a href="https://github.com/GinoBonazza/DCM_snRNAseq">
    <span class="fab fa-github"></span>
     
    Source code
  </a>
</li>
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div id="header">



<h1 class="title toc-ignore">Differential expression analysis in all
cell types using mPAP</h1>
<h4 class="author">GinoBonazza (<a
href="mailto:ginoandrea.bonazza@usz.ch"
class="email">ginoandrea.bonazza@usz.ch</a>)</h4>
<h4 class="date">25 September, 2024</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span>
workflowr <span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span>
</a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2024-09-25
</p>
<p>
<strong>Checks:</strong> <span
class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 7
<span class="glyphicon glyphicon-exclamation-sign text-danger"
aria-hidden="true"></span> 0
</p>
<p>
<strong>Knit directory:</strong> <code>DCM_snRNAseq/</code> <span
class="glyphicon glyphicon-question-sign" aria-hidden="true"
title="This is the local directory in which the code in this file was executed.">
</span>
</p>
<p>
This reproducible <a href="https://rmarkdown.rstudio.com">R Markdown</a>
analysis was created with <a
  href="https://github.com/workflowr/workflowr">workflowr</a> (version
1.7.1). The <em>Checks</em> tab describes the reproducibility checks
that were applied when the results were created. The <em>Past
versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date
</a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git
repository, you know the exact version of the code that produced these
results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the
global environment can affect the analysis in your R Markdown file in
unknown ways. For reproduciblity it’s best to always run the code in an
empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20240606code">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Seed:</strong>
<code>set.seed(20240606)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20240606code"
class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20240606)</code> was run prior to running
the code in the R Markdown file. Setting a seed ensures that any results
that rely on randomness, e.g. subsampling or permutations, are
reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Session information:</strong>
recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded"
class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package
versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be
confident that you successfully produced the results during this
run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongrelative">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>File paths:</strong> relative </a>
</p>
</div>
<div id="strongFilepathsstrongrelative" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Using relative paths to the files within your workflowr
project makes it easier to run your code on other machines.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomGinoBonazzaDCMsnRNAseqtree308c4a111be26da6a7561d87b638a1ce394d6bc7targetblank308c4a1a">
<span class="glyphicon glyphicon-ok text-success"
aria-hidden="true"></span> <strong>Repository version:</strong>
<a href="https://github.com/GinoBonazza/DCM_snRNAseq/tree/308c4a111be26da6a7561d87b638a1ce394d6bc7" target="_blank">308c4a1</a>
</a>
</p>
</div>
<div
id="strongRepositoryversionstrongahrefhttpsgithubcomGinoBonazzaDCMsnRNAseqtree308c4a111be26da6a7561d87b638a1ce394d6bc7targetblank308c4a1a"
class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development
and connecting the code version to the results is critical for
reproducibility.
</p>
<p>
The results in this page were generated with repository version
<a href="https://github.com/GinoBonazza/DCM_snRNAseq/tree/308c4a111be26da6a7561d87b638a1ce394d6bc7" target="_blank">308c4a1</a>.
See the <em>Past versions</em> tab to see a history of the changes made
to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for
the analysis have been committed to Git prior to generating the results
(you can use <code>wflow_publish</code> or
<code>wflow_git_commit</code>). workflowr only checks the R Markdown
file, but you know if there are other scripts or data files that it
depends on. Below is the status of the Git repository when the results
were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    output/Comparison_RV_bulk_RNAseq/
    Ignored:    output/SSc_PAH/

Untracked files:
    Untracked:  GRCh38-2020-A_build/
    Untracked:  analysis/Comparison_RV_bulk_RNAseq.Rmd
    Untracked:  analysis/DA_DE_all_cell_types_and_parameters.Rmd
    Untracked:  analysis/DE_all_cell_types_mPAP.Rmd
    Untracked:  analysis/VennDiagram.2024-09-16_14-12-56.160498.log
    Untracked:  analysis/VennDiagram.2024-09-16_14-13-17.340014.log
    Untracked:  analysis/VennDiagram.2024-09-16_14-13-57.942673.log
    Untracked:  analysis/VennDiagram.2024-09-16_14-22-50.621507.log
    Untracked:  analysis/VennDiagram.2024-09-16_14-24-47.010131.log
    Untracked:  analysis/VennDiagram.2024-09-18_11-22-29.278827.log
    Untracked:  analysis/omnipathr-log/
    Untracked:  code/Add_metadata.R
    Untracked:  code/Check DYSF.R
    Untracked:  code/Clustering_genes.R
    Untracked:  code/DE_5_percent.Rmd
    Untracked:  code/DE_5_percent.html
    Untracked:  code/DE_5_percent/
    Untracked:  code/DE_CM_test1.R
    Untracked:  code/DE_no_401.Rmd
    Untracked:  code/DE_no_401.html
    Untracked:  code/DE_no_401/
    Untracked:  code/Differential abundance test1.R
    Untracked:  code/Differential_Expression_edgeR_All.Rmd
    Untracked:  code/Differential_Expression_edgeR_All_2.Rmd
    Untracked:  code/Differential_Expression_edgeR_All_2.html
    Untracked:  code/Differential_Expression_edgeR_All_Age.Rmd
    Untracked:  code/Differential_Expression_edgeR_All_Age_2.Rmd
    Untracked:  code/Differential_Expression_edgeR_All_Age_2.html
    Untracked:  code/Differential_Expression_edgeR_All_groups.Rmd
    Untracked:  code/Differential_Expression_edgeR_All_groups.html
    Untracked:  code/Differential_Expression_edgeR_All_groups_2.Rmd
    Untracked:  code/Differential_Expression_edgeR_All_groups_2.html
    Untracked:  code/Differential_Expression_edgeR_All_groups_3.Rmd
    Untracked:  code/Differential_Expression_edgeR_All_groups_3.html
    Untracked:  code/PCA and CCA analysis test.R
    Untracked:  code/Pseudobulk DCM HC.R
    Untracked:  code/Published_heart_datasets_RV_vs_LV.Rmd
    Untracked:  code/QC_integration_annotation_orig.Rmd
    Untracked:  code/UpSet_plot_DEGs.Rmd
    Untracked:  code/Volcano_highlighted_genes.R
    Untracked:  code/ezInteractiveTable.Rmd
    Untracked:  code/ezInteractiveTable.html
    Untracked:  code/old.R
    Untracked:  core
    Untracked:  data/Cellbender_output/
    Untracked:  data/Cellranger_output/
    Untracked:  data/DCM_Clinical_data.xlsx
    Untracked:  data/DCM_Clinical_data_26.xlsx
    Untracked:  data/Homo_sapiens.GRCh38.93.gtf
    Untracked:  data/Human_plasma_proteome.xlsx
    Untracked:  data/Published_datasets/
    Untracked:  data/Raw/
    Untracked:  data/SSc-PAH/
    Untracked:  data/gencode.v46.chr_patch_hapl_scaff.annotation.gtf
    Untracked:  data/gencode.v46.chr_patch_hapl_scaff.basic.annotation.gtf
    Untracked:  data/refdata-gex-GRCh38-2020-A/
    Untracked:  omnipathr-log/
    Untracked:  output/Cardiomyocytes_DA_DE/
    Untracked:  output/Cardiomyocytes_DA_DE_26/
    Untracked:  output/Cardiomyocytes_subclustering/
    Untracked:  output/Cardiomyocytes_subclustering_26/
    Untracked:  output/Comparison_LV/
    Untracked:  output/DA_DE_all_cell_types_and_parameters/
    Untracked:  output/DE_all_cell_types_mPAP/
    Untracked:  output/DE_mPAP_all_cell_types/
    Untracked:  output/Differential_expression_edgeR_All/
    Untracked:  output/Differential_expression_edgeR_All_Age/
    Untracked:  output/QC_integration_annotation/
    Untracked:  output/QC_integration_annotation_26/
    Untracked:  output/Reichart_DE_DCMvsHC/
    Untracked:  output/UntitledR.R
    Untracked:  reference_sources/

Unstaged changes:
    Modified:   analysis/Cardiomyocytes_DA_DE.Rmd
    Deleted:    analysis/Differential_Expression_edgeR_All.Rmd
    Deleted:    analysis/Differential_Expression_edgeR_All_Age.Rmd
    Modified:   analysis/SSc_PAH.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not
included in this status report because it is ok for generated content to
have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were
made to the R Markdown
(<code>analysis/DE_mPAP_all_cell_types.Rmd</code>) and HTML
(<code>docs/DE_mPAP_all_cell_types.html</code>) files. If you’ve
configured a remote Git repository (see <code>?wflow_git_remote</code>),
click on the hyperlinks in the table below to view the files as they
were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/GinoBonazza/DCM_snRNAseq/blob/839c8cbcb6bd791307cf90e72852183f3a31434f/analysis/DE_mPAP_all_cell_types.Rmd" target="_blank">839c8cb</a>
</td>
<td>
GinoBonazza
</td>
<td>
2024-09-23
</td>
<td>
wflow_publish("analysis/DE_mPAP_all_cell_types.Rmd")
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="setup" class="section level2">
<h2>Setup</h2>
<pre class="r fold-hide"><code># Get current file name to make folder
current_file &lt;- &quot;DE_mPAP_all_cell_types&quot;

# Load libraries
library(here)
library(readr)
library(readxl)
library(xlsx)
library(Seurat)
library(DropletUtils)
library(Matrix)
library(scDblFinder)
library(scCustomize)
library(dplyr)
library(ggplot2)
library(magrittr)
library(tidyverse)
library(reshape2)
library(S4Vectors)
library(SingleCellExperiment)
library(pheatmap)
library(png)
library(gridExtra)
library(knitr)
library(scales)
library(RColorBrewer)
library(Matrix.utils)
library(tibble)
library(ggplot2)
library(scater)
library(patchwork)
library(statmod)
library(ArchR)
library(clustree)
library(harmony)
library(gprofiler2)
library(clusterProfiler)
library(org.Hs.eg.db)
library(AnnotationHub)
library(ReactomePA)
library(statmod)
library(edgeR)
library(speckle)
library(EnhancedVolcano)
library(decoupleR)
library(OmnipathR)
library(dorothea)
library(enrichplot)
library(png)
library(reactable)
library(UpSetR)
library(ComplexHeatmap)
library(rtracklayer)
library(cowplot)

#Output paths
output_dir_data &lt;- here::here(&quot;output&quot;, current_file)
if (!dir.exists(output_dir_data)) dir.create(output_dir_data)

if (!dir.exists(here::here(&quot;docs&quot;, &quot;figure&quot;))) dir.create(here::here(&quot;docs&quot;, &quot;figure&quot;))

output_dir_figs &lt;- here::here(&quot;docs&quot;, &quot;figure&quot;, paste0(current_file, &quot;.Rmd&quot;))
if (!dir.exists(output_dir_figs)) dir.create(output_dir_figs)</code></pre>
</div>
<div id="de-mpap" class="section level2">
<h2>DE mPAP</h2>
<p>Load seurat object</p>
<pre class="r"><code>DCM_RV_integrated &lt;- readRDS(here::here(&quot;output&quot;, &quot;QC_integration_annotation_26&quot;, &quot;DCM_RV_integrated.rds&quot;))</code></pre>
<p>Load DE results</p>
<pre class="r"><code>cluster_names &lt;- c(&quot;Cardiomyocytes&quot;, &quot;Fibroblasts&quot;, &quot;Endothelial_cells&quot;, &quot;Pericytes&quot;, &quot;Macrophages&quot;, &quot;Lymphocytes&quot;, &quot;Smooth_muscle_cells&quot;, &quot;Neuronal_cells&quot;, &quot;Endocardial_cells&quot;)</code></pre>
<pre class="r"><code>results &lt;- list()
for (i in seq_along(cluster_names)) {
  results[[i]] &lt;- read.csv(here::here(&quot;output&quot;, &quot;DA_DE_all_cell_types_and_parameters&quot;, paste0(cluster_names[i], &quot;_DE_Results_mPAP_mmHg.csv&quot;)))
  names(results)[i] &lt;- cluster_names[i]
}</code></pre>
<pre class="r"><code>signif &lt;- list()
for (i in seq_along(cluster_names)) {
  signif[[i]] &lt;- read.csv(here::here(&quot;output&quot;, &quot;DA_DE_all_cell_types_and_parameters&quot;, paste0(cluster_names[i], &quot;_DE_Significant_mPAP_mmHg.csv&quot;)))
  names(signif)[i] &lt;- cluster_names[i]
}</code></pre>
<p>Add info to the results data frame: coding/not coding and
unique/overlapping</p>
<pre class="r"><code>reference_genes &lt;- readGFF(here::here(&quot;data&quot;, &quot;refdata-gex-GRCh38-2020-A&quot;, &quot;genes&quot;, &quot;genes.gtf&quot;))
reference_genes &lt;- unique(dplyr::select(reference_genes, gene_id, gene_type, gene_name))</code></pre>
<pre class="r"><code>for (i in seq_along(results)) {
  results[[i]] &lt;- left_join(results[[i]], reference_genes, by = c(&quot;gene&quot; = &quot;gene_name&quot;))
}</code></pre>
<pre class="r"><code>for (i in seq_along(results)) {
  results[[i]]$up_down &lt;- ifelse(
    results[[i]]$logFC &gt; 0.5 &amp; results[[i]]$FDR &lt; 0.05, &quot;Increased&quot;,
    ifelse(results[[i]]$logFC &lt; -0.5 &amp; results[[i]]$FDR &lt; 0.05, &quot;Decreased&quot;,
           &quot;Not significant&quot;
           )
  )
}</code></pre>
<pre class="r"><code>for (i in seq_along(results)) {
  others_up &lt;- unlist(lapply(signif[-i], function(df) df$gene[df$logFC &gt; 0.5]))
  others_down &lt;- unlist(lapply(signif[-i], function(df) df$gene[df$logFC &lt; -0.5]))
  results[[i]]$unique_overlapping &lt;- ifelse(
    results[[i]]$up_down == &quot;Increased&quot; &amp; results[[i]]$gene %in% others_up, &quot;Overlapping&quot;,
    ifelse(results[[i]]$up_down == &quot;Decreased&quot; &amp; results[[i]]$gene %in% others_down, &quot;Overlapping&quot;,
           ifelse(results[[i]]$up_down == &quot;Increased&quot; &amp; !results[[i]]$gene %in% others_up, &quot;Unique&quot;,
                  ifelse(results[[i]]$up_down == &quot;Decreased&quot; &amp; !results[[i]]$gene %in% others_down, &quot;Unique&quot;,
                         &quot;Not significant&quot;
                  )
           )
    )
  )
}</code></pre>
<p>Volcano plots</p>
<pre class="r"><code>get_legend &lt;- function(my_plot) {
  tmp &lt;- ggplotGrob(my_plot)
  legend &lt;- tmp$grobs[[which(sapply(tmp$grobs, function(x) x$name) == &quot;guide-box&quot;)]]
  return(legend)
}</code></pre>
<pre class="r"><code>volcano &lt;- list()
for (i in seq_along(results)) {
  keyvals &lt;- rep(&quot;grey&quot;, nrow(results[[i]]))
  names(keyvals) &lt;- rep(&quot;Not significant&quot;, nrow(results[[i]]))
  
  keyvals[results[[i]]$up_down == &quot;Increased&quot;] &lt;- &quot;#A63A2A&quot;
  names(keyvals)[keyvals == &quot;#A63A2A&quot;] &lt;- &quot;Increased&quot;
  keyvals[results[[i]]$up_down == &quot;Decreased&quot;] &lt;- &quot;#004C8C&quot;
  names(keyvals)[keyvals == &quot;#004C8C&quot;] &lt;- &quot;Decreased&quot;

  volcano[[i]] &lt;- EnhancedVolcano(results[[i]],
                  lab = results[[i]]$gene,
                  x = &quot;logFC&quot;,
                  y = &quot;FDR&quot;,
                  labSize = 0,
                  legendLabSize = 15,
                  titleLabSize = 16,
                  subtitleLabSize = 14,
                  axisLabSize = 12,
                  captionLabSize = 9,
                  pointSize = 1,
                  FCcutoff = 0.5,
                  pCutoff  = 0.05,
                  ylim = c(0, 3.5),
                  colAlpha = 1,
                  drawConnectors = FALSE,
                  subtitle = NULL,
                  title = names(results)[i],
                  colCustom = keyvals
  ) + theme(legend.position = &quot;none&quot;)
  names(volcano)[i] &lt;- names(results)[i]
}

# Extract the legend from the first plot
legend &lt;- cowplot::get_legend(volcano[[1]] + theme(legend.position = &quot;right&quot;))

# Combine the individual volcano plots
combined_plot &lt;- patchwork::wrap_plots(volcano, ncol = 5)

# Use ggdraw and draw_plot to overlay the legend in the bottom-right corner
final_plot &lt;- cowplot::ggdraw() +
  cowplot::draw_plot(combined_plot, 0, 0, 1, 1) +    # Place the combined plot to fill the whole canvas
  cowplot::draw_plot(legend, 0.77, 0.15, 0.25, 0.25)    # Position the legend in the bottom-right

# Print the final plot
print(final_plot)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/Volcano_up_down-1.png" width="4800" style="display: block; margin: auto;" /></p>
<pre class="r"><code>volcano &lt;- list()
for (i in seq_along(results)) {
  keyvals &lt;- rep(&quot;grey&quot;, nrow(results[[i]]))
  names(keyvals) &lt;- rep(&quot;Not significant&quot;, nrow(results[[i]]))
  
  keyvals[results[[i]]$up_down == &quot;Increased&quot; &amp; results[[i]]$gene_type == &quot;protein_coding&quot;] &lt;- &quot;#A63A2A&quot;
  names(keyvals)[keyvals == &quot;#A63A2A&quot;] &lt;- &quot;Increased - protein coding&quot;
  keyvals[results[[i]]$up_down == &quot;Decreased&quot; &amp; results[[i]]$gene_type == &quot;protein_coding&quot;] &lt;- &quot;#004C8C&quot;
  names(keyvals)[keyvals == &quot;#004C8C&quot;] &lt;- &quot;Decreased - protein coding&quot;
  keyvals[results[[i]]$up_down == &quot;Increased&quot; &amp; results[[i]]$gene_type == &quot;lncRNA&quot;] &lt;- &quot;#228B22&quot;
  names(keyvals)[keyvals == &quot;#228B22&quot;] &lt;- &quot;Increased - non coding&quot;
  keyvals[results[[i]]$up_down == &quot;Decreased&quot; &amp; results[[i]]$gene_type == &quot;lncRNA&quot;] &lt;- &quot;#FFCC00&quot;
  names(keyvals)[keyvals == &quot;#FFCC00&quot;] &lt;- &quot;Decreased - non coding&quot;

  volcano[[i]] &lt;- EnhancedVolcano(results[[i]],
                  lab = results[[i]]$gene,
                  x = &quot;logFC&quot;,
                  y = &quot;FDR&quot;,
                  labSize = 0,
                  legendLabSize = 15,
                  titleLabSize = 16,
                  subtitleLabSize = 14,
                  axisLabSize = 12,
                  captionLabSize = 9,
                  pointSize = 1,
                  FCcutoff = 0.5,
                  pCutoff  = 0.05,
                  ylim = c(0, 3.5),
                  colAlpha = 1,
                  drawConnectors = FALSE,
                  subtitle = NULL,
                  title = names(results)[i],
                  colCustom = keyvals
  ) + theme(legend.position = &quot;none&quot;)
  names(volcano)[i] &lt;- names(results)[i]
}

# Extract the legend from the first plot
legend &lt;- cowplot::get_legend(volcano[[1]] + theme(legend.position = &quot;right&quot;))

# Combine the individual volcano plots
combined_plot &lt;- patchwork::wrap_plots(volcano, ncol = 5)

# Use ggdraw and draw_plot to overlay the legend in the bottom-right corner
final_plot &lt;- cowplot::ggdraw() +
  cowplot::draw_plot(combined_plot, 0, 0, 1, 1) +    # Place the combined plot to fill the whole canvas
  cowplot::draw_plot(legend, 0.77, 0.15, 0.25, 0.25)    # Position the legend in the bottom-right

# Print the final plot
print(final_plot)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/Volcano_gene_type-1.png" width="4800" style="display: block; margin: auto;" /></p>
<pre class="r"><code>volcano &lt;- list()
for (i in seq_along(results)) {
  keyvals &lt;- rep(&quot;grey&quot;, nrow(results[[i]]))
  names(keyvals) &lt;- rep(&quot;Not significant&quot;, nrow(results[[i]]))
  
  keyvals[results[[i]]$up_down == &quot;Increased&quot; &amp; results[[i]]$unique_overlapping == &quot;Unique&quot;] &lt;- &quot;#A63A2A&quot;
  names(keyvals)[keyvals == &quot;#A63A2A&quot;] &lt;- &quot;Increased - Unique&quot;
  keyvals[results[[i]]$up_down == &quot;Decreased&quot; &amp; results[[i]]$unique_overlapping == &quot;Unique&quot;] &lt;- &quot;#004C8C&quot;
  names(keyvals)[keyvals == &quot;#004C8C&quot;] &lt;- &quot;Decreased - Unique&quot;
  keyvals[results[[i]]$up_down == &quot;Increased&quot; &amp; results[[i]]$unique_overlapping == &quot;Overlapping&quot;] &lt;- &quot;#D6A0A0&quot;
  names(keyvals)[keyvals == &quot;#D6A0A0&quot;] &lt;- &quot;Increased - Overlapping&quot;
  keyvals[results[[i]]$up_down == &quot;Decreased&quot; &amp; results[[i]]$unique_overlapping == &quot;Overlapping&quot;] &lt;- &quot;#A0C4FF&quot;
  names(keyvals)[keyvals == &quot;#A0C4FF&quot;] &lt;- &quot;Decreased - Overlapping&quot;

  volcano[[i]] &lt;- EnhancedVolcano(results[[i]],
                  lab = results[[i]]$gene,
                  x = &quot;logFC&quot;,
                  y = &quot;FDR&quot;,
                  labSize = 0,
                  legendLabSize = 15,
                  titleLabSize = 16,
                  subtitleLabSize = 14,
                  axisLabSize = 12,
                  captionLabSize = 9,
                  pointSize = 1,
                  FCcutoff = 0.5,
                  pCutoff  = 0.05,
                  ylim = c(0, 3.5),
                  colAlpha = 1,
                  drawConnectors = FALSE,
                  subtitle = NULL,
                  title = names(results)[i],
                  colCustom = keyvals
  ) + theme(legend.position = &quot;none&quot;)
  names(volcano)[i] &lt;- names(results)[i]
}

# Extract the legend from the first plot
legend &lt;- cowplot::get_legend(volcano[[1]] + theme(legend.position = &quot;right&quot;))

# Combine the individual volcano plots
combined_plot &lt;- patchwork::wrap_plots(volcano, ncol = 5)

# Use ggdraw and draw_plot to overlay the legend in the bottom-right corner
final_plot &lt;- cowplot::ggdraw() +
  cowplot::draw_plot(combined_plot, 0, 0, 1, 1) +    # Place the combined plot to fill the whole canvas
  cowplot::draw_plot(legend, 0.77, 0.15, 0.25, 0.25)    # Position the legend in the bottom-right

# Print the final plot
print(final_plot)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/Volcano_unique_overlapping-1.png" width="4800" style="display: block; margin: auto;" /></p>
<pre class="r"><code>volcano &lt;- list()
results_no_lnc &lt;- results
for (i in seq_along(results)) {
  results_no_lnc[[i]] &lt;- dplyr::filter(results_no_lnc[[i]], gene_type == &quot;protein_coding&quot;)
  keyvals &lt;- rep(&quot;grey&quot;, nrow(results_no_lnc[[i]]))
  names(keyvals) &lt;- rep(&quot;Not significant&quot;, nrow(results_no_lnc[[i]]))
  
  keyvals[results_no_lnc[[i]]$up_down == &quot;Increased&quot; &amp; results_no_lnc[[i]]$gene_type == &quot;protein_coding&quot;] &lt;- &quot;#A63A2A&quot;
  names(keyvals)[keyvals == &quot;#A63A2A&quot;] &lt;- &quot;Increased - protein coding&quot;
  keyvals[results_no_lnc[[i]]$up_down == &quot;Decreased&quot; &amp; results_no_lnc[[i]]$gene_type == &quot;protein_coding&quot;] &lt;- &quot;#004C8C&quot;
  names(keyvals)[keyvals == &quot;#004C8C&quot;] &lt;- &quot;Decreased - protein coding&quot;

  volcano[[i]] &lt;- EnhancedVolcano(results_no_lnc[[i]],
                  lab = results_no_lnc[[i]]$gene,
                  x = &quot;logFC&quot;,
                  y = &quot;FDR&quot;,
                  labSize = 0,
                  legendLabSize = 15,
                  titleLabSize = 16,
                  subtitleLabSize = 14,
                  axisLabSize = 12,
                  captionLabSize = 9,
                  pointSize = 1,
                  FCcutoff = 0.5,
                  pCutoff  = 0.05,
                  ylim = c(0, 3.5),
                  colAlpha = 1,
                  drawConnectors = FALSE,
                  subtitle = NULL,
                  title = names(results_no_lnc)[i],
                  colCustom = keyvals
  ) + theme(legend.position = &quot;none&quot;)
  names(volcano)[i] &lt;- names(results_no_lnc)[i]
}

# Extract the legend from the first plot
legend &lt;- cowplot::get_legend(volcano[[1]] + theme(legend.position = &quot;right&quot;))

# Combine the individual volcano plots
combined_plot &lt;- patchwork::wrap_plots(volcano, ncol = 5)

# Use ggdraw and draw_plot to overlay the legend in the bottom-right corner
final_plot &lt;- cowplot::ggdraw() +
  cowplot::draw_plot(combined_plot, 0, 0, 1, 1) +    # Place the combined plot to fill the whole canvas
  cowplot::draw_plot(legend, 0.77, 0.15, 0.25, 0.25)    # Position the legend in the bottom-right

# Print the final plot
print(final_plot)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/Volcano_up_down_no_lnc-1.png" width="4800" style="display: block; margin: auto;" /></p>
<pre class="r"><code>volcano &lt;- list()
results_no_overlap &lt;- results
for (i in seq_along(results)) {
  results_no_overlap[[i]] &lt;- dplyr::filter(results_no_overlap[[i]], unique_overlapping %in% c(&quot;Unique&quot;, &quot;Not significant&quot;))
  keyvals &lt;- rep(&quot;grey&quot;, nrow(results_no_overlap[[i]]))
  names(keyvals) &lt;- rep(&quot;Not significant&quot;, nrow(results_no_overlap[[i]]))
  
  keyvals[results_no_overlap[[i]]$up_down == &quot;Increased&quot; &amp; results_no_overlap[[i]]$unique_overlapping == &quot;Unique&quot;] &lt;- &quot;#A63A2A&quot;
  names(keyvals)[keyvals == &quot;#A63A2A&quot;] &lt;- &quot;Increased - Unique&quot;
  keyvals[results_no_overlap[[i]]$up_down == &quot;Decreased&quot; &amp; results_no_overlap[[i]]$unique_overlapping == &quot;Unique&quot;] &lt;- &quot;#004C8C&quot;
  names(keyvals)[keyvals == &quot;#004C8C&quot;] &lt;- &quot;Decreased - Unique&quot;

  volcano[[i]] &lt;- EnhancedVolcano(results_no_overlap[[i]],
                  lab = results_no_overlap[[i]]$gene,
                  x = &quot;logFC&quot;,
                  y = &quot;FDR&quot;,
                  labSize = 0,
                  legendLabSize = 15,
                  titleLabSize = 16,
                  subtitleLabSize = 14,
                  axisLabSize = 12,
                  captionLabSize = 9,
                  pointSize = 1,
                  FCcutoff = 0.5,
                  pCutoff  = 0.05,
                  ylim = c(0, 3.5),
                  colAlpha = 1,
                  drawConnectors = FALSE,
                  subtitle = NULL,
                  title = names(results_no_overlap)[i],
                  colCustom = keyvals
  ) + theme(legend.position = &quot;none&quot;)
  names(volcano)[i] &lt;- names(results_no_overlap)[i]
}

# Extract the legend from the first plot
legend &lt;- cowplot::get_legend(volcano[[1]] + theme(legend.position = &quot;right&quot;))

# Combine the individual volcano plots
combined_plot &lt;- patchwork::wrap_plots(volcano, ncol = 5)

# Use ggdraw and draw_plot to overlay the legend in the bottom-right corner
final_plot &lt;- cowplot::ggdraw() +
  cowplot::draw_plot(combined_plot, 0, 0, 1, 1) +    # Place the combined plot to fill the whole canvas
  cowplot::draw_plot(legend, 0.77, 0.15, 0.25, 0.25)    # Position the legend in the bottom-right

# Print the final plot
print(final_plot)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/Volcano_up_down_no_overlap-1.png" width="4800" style="display: block; margin: auto;" /></p>
<p>Bar plots</p>
<pre class="r"><code>signif &lt;- results
for (i in seq_along(results)) {
  signif[[i]] &lt;- results[[i]] %&gt;%
    dplyr::filter(FDR &lt; 0.05) %&gt;%
    dplyr::filter(abs(logFC) &gt; 0.5)
}</code></pre>
<pre class="r"><code>n_de_genes &lt;- data.frame()
for (i in seq_along(signif)) {
  cell_type &lt;- c(rep(names(signif)[i], 2))
  effect &lt;- c(&quot;Increased&quot;, &quot;Decreased&quot;)
  n &lt;- c(nrow(dplyr::filter(signif[[i]], logFC &gt; 0)), nrow(dplyr::filter(signif[[i]], logFC &lt; 0)))
  de_tot &lt;- nrow(signif[[i]])
  n_de_genes &lt;- rbind(n_de_genes, data.frame(cell_type,effect,n, de_tot))
}
n_de_genes &lt;- n_de_genes %&gt;%
  arrange(desc(de_tot))
n_de_genes$cell_type &lt;- factor(n_de_genes$cell_type, levels = unique(n_de_genes$cell_type))

# Plot with customizations
ggplot(n_de_genes, aes(fill=effect, y=n, x=cell_type)) + 
  geom_bar(position=&quot;stack&quot;, stat=&quot;identity&quot;) +
  scale_fill_manual(values = c(&quot;Increased&quot; = &quot;#A63A2A&quot;, &quot;Decreased&quot; = &quot;#004C8C&quot;)) +
  theme(text = element_text(size = 15), axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = &quot;Cell Type&quot;, y = &quot;Number of Genes&quot;, fill = &quot;Effect&quot;) +
  guides(fill = guide_legend(title = NULL))</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/Barplot_up_down-1.png" width="2700" style="display: block; margin: auto;" /></p>
<pre class="r"><code>n_de_genes &lt;- data.frame()
for (i in seq_along(signif)) {
  cell_type &lt;- names(signif)[i]
  effect &lt;- c(&quot;Increased - protein coding&quot;, &quot;Increased - non coding&quot;, &quot;Decreased - protein coding&quot;, &quot;Decreased - non coding&quot;)
  n &lt;- c(nrow(dplyr::filter(signif[[i]], logFC &gt; 0 &amp; gene_type == &quot;protein_coding&quot;)),
         nrow(dplyr::filter(signif[[i]], logFC &gt; 0 &amp; gene_type == &quot;lncRNA&quot;)),
         nrow(dplyr::filter(signif[[i]], logFC &lt; 0 &amp; gene_type == &quot;protein_coding&quot;)),
         nrow(dplyr::filter(signif[[i]], logFC &lt; 0 &amp; gene_type == &quot;lncRNA&quot;))
         )
  de_tot &lt;- nrow(signif[[i]])
  n_de_genes &lt;- rbind(n_de_genes, data.frame(cell_type,effect,n, de_tot))
}
n_de_genes$cell_type &lt;- factor(n_de_genes$cell_type, levels = unique(n_de_genes$cell_type))

ggplot(n_de_genes, aes(fill=effect, y=n, x=cell_type)) + 
  geom_bar(position=&quot;stack&quot;, stat=&quot;identity&quot;) +
  scale_fill_manual(values = c(&quot;Increased - protein coding&quot; = &quot;#A63A2A&quot;, 
                               &quot;Decreased - protein coding&quot; = &quot;#004C8C&quot;, 
                               &quot;Increased - non coding&quot; = &quot;#D6A0A0&quot;, 
                               &quot;Decreased - non coding&quot; = &quot;#A0C4FF&quot;)) +
  theme(text = element_text(size = 15), axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = &quot;Cell Type&quot;, y = &quot;Number of Genes&quot;, fill = &quot;Effect&quot;) +
  guides(fill = guide_legend(title = NULL))</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/Barplot_gene_type-1.png" width="2700" style="display: block; margin: auto;" /></p>
<pre class="r"><code>n_de_genes&lt;- data.frame()
for (i in seq_along(signif)) {
  cell_type &lt;- names(signif)[i]
  effect &lt;- c(&quot;Increased - Unique&quot;, &quot;Increased - Overlapping&quot;, &quot;Decreased - Unique&quot;, &quot;Decreased - Overlapping&quot;)
  n &lt;- c(nrow(dplyr::filter(signif[[i]], logFC &gt; 0 &amp; unique_overlapping == &quot;Unique&quot;)),
         nrow(dplyr::filter(signif[[i]], logFC &gt; 0 &amp; unique_overlapping == &quot;Overlapping&quot;)),
         nrow(dplyr::filter(signif[[i]], logFC &lt; 0 &amp; unique_overlapping == &quot;Unique&quot;)),
         nrow(dplyr::filter(signif[[i]], logFC &lt; 0 &amp; unique_overlapping == &quot;Overlapping&quot;))
         )
  de_tot &lt;- nrow(signif[[i]])
  n_de_genes &lt;- rbind(n_de_genes, data.frame(cell_type,effect,n, de_tot))
}
n_de_genes$cell_type &lt;- factor(n_de_genes$cell_type, levels = unique(n_de_genes$cell_type))

ggplot(n_de_genes, aes(fill=effect, y=n, x=cell_type)) + 
  geom_bar(position=&quot;stack&quot;, stat=&quot;identity&quot;) +
  scale_fill_manual(values = c(&quot;Increased - Unique&quot; = &quot;#A63A2A&quot;, 
                               &quot;Decreased - Unique&quot; = &quot;#004C8C&quot;, 
                               &quot;Increased - Overlapping&quot; = &quot;#D6A0A0&quot;, 
                               &quot;Decreased - Overlapping&quot; = &quot;#A0C4FF&quot;)) +
  theme(text = element_text(size = 15), axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = &quot;Cell Type&quot;, y = &quot;Number of Genes&quot;, fill = &quot;Effect&quot;) +
  guides(fill = guide_legend(title = NULL))</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/Barplot_up_down_unique_overlapping-1.png" width="2700" style="display: block; margin: auto;" /></p>
<pre class="r"><code>n_de_genes&lt;- data.frame()
for (i in seq_along(signif)) {
  cell_type &lt;- names(signif)[i]
  effect &lt;- c(&quot;Unique&quot;, &quot;Overlapping&quot;)
  n &lt;- c(nrow(dplyr::filter(signif[[i]], unique_overlapping == &quot;Unique&quot;)),
         nrow(dplyr::filter(signif[[i]], unique_overlapping == &quot;Overlapping&quot;))
         )
  de_tot &lt;- nrow(signif[[i]])
  n_de_genes &lt;- rbind(n_de_genes, data.frame(cell_type,effect,n, de_tot))
}
n_de_genes$cell_type &lt;- factor(n_de_genes$cell_type, levels = unique(n_de_genes$cell_type))

ggplot(n_de_genes, aes(fill=effect, y=n, x=cell_type)) + 
  geom_bar(position=&quot;stack&quot;, stat=&quot;identity&quot;) +
  scale_fill_manual(values = c(&quot;Unique&quot; = &quot;#A63A2A&quot;, 
                               &quot;Overlapping&quot; = &quot;#004C8C&quot;)) +
  theme(text = element_text(size = 15), axis.text.x = element_text(angle = 45, hjust = 1)) +
  labs(x = &quot;Cell Type&quot;, y = &quot;Number of Genes&quot;, fill = &quot;Effect&quot;) +
  guides(fill = guide_legend(title = NULL))</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/Barplot_unique_overlapping-1.png" width="2700" style="display: block; margin: auto;" /></p>
<p>UpSet plots</p>
<pre class="r"><code>signif_up &lt;- signif
for (i in seq_along(signif_up)) {
  signif_up[[i]] &lt;- filter(signif_up[[i]], logFC &gt; 0)
}

signif_down &lt;- signif
for (i in seq_along(signif_down)) {
  signif_down[[i]] &lt;- filter(signif_down[[i]], logFC &lt; 0)
}</code></pre>
<pre class="r"><code>gene_sets &lt;- list()
for (i in seq_along(signif_up)) {
  gene_sets[[i]] &lt;- signif_up[[i]]$gene
  names(gene_sets)[i] &lt;- names(signif_up)[i]
  }
m &lt;- make_comb_mat(gene_sets, mode = &quot;distinct&quot;, min_set_size = 10)
#m &lt;- m[comb_degree(m) &gt;= 2]
m &lt;- m[comb_size(m) &gt;= 10]
p1 &lt;- UpSet(m, column_title = &quot;DE mPAP_mmHg - Upregulated&quot;, comb_order = order(comb_size(m)), 
                    top_annotation = upset_top_annotation(m, add_numbers = TRUE),
                    right_annotation = upset_right_annotation(m, add_numbers = TRUE))
gene_sets &lt;- list()
for (i in seq_along(signif_down)) {
  gene_sets[[i]] &lt;- signif_down[[i]]$gene
  names(gene_sets)[i] &lt;- names(signif_down)[i]
  }
m &lt;- make_comb_mat(gene_sets, mode = &quot;distinct&quot;, min_set_size = 10)
#m &lt;- m[comb_degree(m) &gt;= 2]
m &lt;- m[comb_size(m) &gt;= 10]
p2 &lt;- UpSet(m, column_title = &quot;DE mPAP_mmHg - Downregulated&quot;, comb_order = order(comb_size(m)), 
                    top_annotation = upset_top_annotation(m, add_numbers = TRUE),
                    right_annotation = upset_right_annotation(m, add_numbers = TRUE))

p1 &lt;- grid::grid.grabExpr(draw(p1))
p2 &lt;- grid::grid.grabExpr(draw(p2))

combined_plots &lt;- plot_grid(p1, p2, ncol = 1)

print(combined_plots)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/UpSet_plot_mPAP-1.png" width="3600" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rm(gene_sets, m, p1, p2, combined_plots)</code></pre>
</div>
<div id="gsea" class="section level2">
<h2>GSEA</h2>
<pre class="r"><code>reorder_GO_by_pvalue &lt;- function(enrichGO_result) {
  go_results &lt;- enrichGO_result@result
  go_results_sorted &lt;- go_results[order(go_results$p.adjust), ]
  enrichGO_result@result &lt;- go_results_sorted
  return(enrichGO_result)
}</code></pre>
<pre class="r"><code>classify_terms &lt;- function(df) {
  df$sign &lt;- ifelse(df$NES &gt; 0, &quot;activated&quot;, &quot;suppressed&quot;)
  return(df)
}</code></pre>
<pre class="r"><code>cluster_names_2 &lt;- c(&quot;Cardiomyocytes&quot;, &quot;Fibroblasts&quot;, &quot;Endothelial cells&quot;, &quot;Pericytes&quot;, &quot;Macrophages&quot;, &quot;Lymphocytes&quot;, &quot;Smooth muscle cells&quot;, &quot;Neuronal cells&quot;, &quot;Endocardial cells&quot;)</code></pre>
<pre class="r"><code>ranked_genes &lt;- list()
GSEA_all &lt;- list()
for (i in seq_along(cluster_names)) {
  gene_counts &lt;- rowSums(subset(DCM_RV_integrated, cell_type == cluster_names_2[i])@assays$RNA@counts &gt; 0)
  universe_genes &lt;- names(gene_counts[gene_counts &gt;= 3])
  write.csv(as.data.frame(universe_genes), here::here(output_dir_data, paste0(cluster_names[i], &quot;_universe_genes.csv&quot;)), quote=F, row.names = F)
  universe_entrez &lt;- bitr(universe_genes, fromType = &quot;SYMBOL&quot;, toType = &quot;ENTREZID&quot;, OrgDb = org.Hs.eg.db)$ENTREZID
  write.csv(as.data.frame(universe_entrez), here::here(output_dir_data, paste0(cluster_names[i], &quot;_universe_entrez.csv&quot;)), quote=F, row.names = F)
  rm(gene_counts)
  
  up_genes &lt;- filter(signif[[i]], logFC &gt; 0.5)$gene
  up_genes &lt;- bitr(up_genes, fromType=&quot;SYMBOL&quot;, toType=&quot;ENTREZID&quot;, OrgDb=&quot;org.Hs.eg.db&quot;)$ENTREZID
  down_genes &lt;- filter(signif[[i]], logFC &lt; -0.5)$gene
  down_genes &lt;- bitr(down_genes, fromType=&quot;SYMBOL&quot;, toType=&quot;ENTREZID&quot;, OrgDb=&quot;org.Hs.eg.db&quot;)$ENTREZID
  
  ranked_genes[[i]] &lt;- results[[i]]
  names(ranked_genes)[i] &lt;- cluster_names[i]
  ranked_genes[[i]]$metric &lt;- ranked_genes[[i]]$logFC*-log10(ranked_genes[[i]]$PValue)
  entrezid &lt;- bitr(ranked_genes[[i]]$gene, fromType=&quot;SYMBOL&quot;, toType=&quot;ENTREZID&quot;, OrgDb=&quot;org.Hs.eg.db&quot;, drop = TRUE)
  ranked_genes[[i]] &lt;- merge(ranked_genes[[i]], entrezid, by.x = &quot;gene&quot;, by.y = &quot;SYMBOL&quot;)
  ranked_genes[[i]] &lt;- ranked_genes[[i]][!duplicated(ranked_genes[[i]]$metric), ]
  write.csv(ranked_genes[[i]], file = here::here(output_dir_data, paste0(cluster_names[i], &quot;_ranked_genes.csv&quot;)), quote=F, row.names = F)
  
  genelist &lt;- ranked_genes[[i]]$metric
  names(genelist) &lt;- ranked_genes[[i]]$ENTREZID
  genelist &lt;- genelist[order(genelist, decreasing = T)]
  
  GSEA_all[i] &lt;- gseGO(geneList = genelist,
                       OrgDb = org.Hs.eg.db,
                       ont = &quot;ALL&quot;,
                       keyType = &quot;ENTREZID&quot;,
                       nPermSimple = 10000000,
                       minGSSize = 10,
                       eps = 0)
  GSEA_all[i] &lt;- reorder_GO_by_pvalue(GSEA_all[i])
  names(GSEA_all)[i] &lt;- cluster_names[i]
  saveRDS(GSEA_all[i], here::here(output_dir_data, paste0(cluster_names[i], &quot;_GSEA_GO_mPAP.rds&quot;)))
}

rm(universe_genes, universe_entrez, up_genes, down_genes, genelist)

saveRDS(ranked_genes, here::here(output_dir_data, &quot;ranked_genes_list.rds&quot;))
saveRDS(GSEA_all, here::here(output_dir_data, &quot;GSEA_all_list.rds&quot;))</code></pre>
<pre class="r"><code>GSEA_all &lt;- readRDS(here::here(output_dir_data, &quot;GSEA_all_list.rds&quot;))</code></pre>
<pre class="r"><code>GSEA_simplified &lt;- simplify(
  GSEA_all[[&quot;Cardiomyocytes&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)

GSEA_simplified &lt;- reorder_GO_by_pvalue(GSEA_simplified)

dotplot(GSEA_simplified, showCategory = 8, split=&quot;.sign&quot;, title = paste0(&quot;Cardiomyocytes&quot;, &quot;\nGO Enrichment Analysis&quot;, &quot;\nmPAP-associated genes&quot;), 
        label_format = 30, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;),
        strip.text = element_text(face = &quot;bold&quot;, size = 18)) +
  facet_grid(.~.sign)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/GSEA_Cardiomyocytes-1.png" width="2400" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GSEA_simplified@result &lt;- classify_terms(GSEA_simplified@result)

activated_terms &lt;- subset(GSEA_simplified@result, sign == &quot;activated&quot;)
suppressed_terms &lt;- subset(GSEA_simplified@result, sign == &quot;suppressed&quot;)

GSEA_activated &lt;- GSEA_simplified
GSEA_activated@result &lt;- activated_terms
GSEA_activated &lt;- pairwise_termsim(GSEA_activated)

GSEA_suppressed &lt;- GSEA_simplified
GSEA_suppressed@result &lt;- suppressed_terms
GSEA_suppressed &lt;- pairwise_termsim(GSEA_suppressed)

p1 &lt;- emapplot(GSEA_activated, showCategory = 100, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Activated Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))

p2 &lt;- emapplot(GSEA_suppressed, showCategory = 100, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Suppressed Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))

combined_plot &lt;- p1 | p2
print(combined_plot)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/GSEA_Cardiomyocytes_emapplot-1.png" width="6000" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rm(GSEA_simplified, activated_terms, suppressed_terms, GSEA_activated, GSEA_suppressed, p1, p2)</code></pre>
<pre class="r"><code>GSEA_simplified &lt;- simplify(
  GSEA_all[[&quot;Fibroblasts&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)

GSEA_simplified &lt;- reorder_GO_by_pvalue(GSEA_simplified)

dotplot(GSEA_simplified, showCategory = 8, split=&quot;.sign&quot;, title = paste0(&quot;Fibroblasts&quot;, &quot;\nGO Enrichment Analysis&quot;, &quot;\nmPAP-associated genes&quot;), 
        label_format = 30, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;),
        strip.text = element_text(face = &quot;bold&quot;, size = 18)) +
  facet_grid(.~.sign)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/GSEA_Fibroblasts-1.png" width="2400" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GSEA_simplified &lt;- simplify(
  GSEA_all[[&quot;Endothelial_cells&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)

GSEA_simplified &lt;- reorder_GO_by_pvalue(GSEA_simplified)

dotplot(GSEA_simplified, showCategory = 8, split=&quot;.sign&quot;, title = paste0(&quot;Endothelial cells&quot;, &quot;\nGO Enrichment Analysis&quot;, &quot;\nmPAP-associated genes&quot;), 
        label_format = 30, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;),
        strip.text = element_text(face = &quot;bold&quot;, size = 18)) +
  facet_grid(.~.sign)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/GSEA_Endothelial_cells-1.png" width="2400" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GSEA_simplified@result &lt;- classify_terms(GSEA_simplified@result)

activated_terms &lt;- subset(GSEA_simplified@result, sign == &quot;activated&quot;)
suppressed_terms &lt;- subset(GSEA_simplified@result, sign == &quot;suppressed&quot;)

GSEA_activated &lt;- GSEA_simplified
GSEA_activated@result &lt;- activated_terms
GSEA_activated &lt;- pairwise_termsim(GSEA_activated)

GSEA_suppressed &lt;- GSEA_simplified
GSEA_suppressed@result &lt;- suppressed_terms
GSEA_suppressed &lt;- pairwise_termsim(GSEA_suppressed)

p1 &lt;- emapplot(GSEA_activated, showCategory = 100, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Activated Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))

p2 &lt;- emapplot(GSEA_suppressed, showCategory = 100, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Suppressed Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))

combined_plot &lt;- p1 | p2
print(combined_plot)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/GSEA_Endothelial_cells_emapplot-1.png" width="6000" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rm(GSEA_simplified, activated_terms, suppressed_terms, GSEA_activated, GSEA_suppressed, p1, p2)</code></pre>
<pre class="r"><code>GSEA_simplified &lt;- simplify(
  GSEA_all[[&quot;Pericytes&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)

GSEA_simplified &lt;- reorder_GO_by_pvalue(GSEA_simplified)

dotplot(GSEA_simplified, showCategory = 8, split=&quot;.sign&quot;, title = paste0(&quot;Pericytes&quot;, &quot;\nGO Enrichment Analysis&quot;, &quot;\nmPAP-associated genes&quot;), 
        label_format = 30, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;),
        strip.text = element_text(face = &quot;bold&quot;, size = 18)) +
  facet_grid(.~.sign)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/GSEA_Pericytes-1.png" width="2400" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GSEA_simplified@result &lt;- classify_terms(GSEA_simplified@result)

activated_terms &lt;- subset(GSEA_simplified@result, sign == &quot;activated&quot;)
suppressed_terms &lt;- subset(GSEA_simplified@result, sign == &quot;suppressed&quot;)

GSEA_activated &lt;- GSEA_simplified
GSEA_activated@result &lt;- activated_terms
GSEA_activated &lt;- pairwise_termsim(GSEA_activated)

GSEA_suppressed &lt;- GSEA_simplified
GSEA_suppressed@result &lt;- suppressed_terms
GSEA_suppressed &lt;- pairwise_termsim(GSEA_suppressed)

p1 &lt;- emapplot(GSEA_activated, showCategory = 100, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Activated Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))

p2 &lt;- emapplot(GSEA_suppressed, showCategory = 100, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Suppressed Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))

combined_plot &lt;- p1 | p2
print(combined_plot)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/GSEA_Pericytes_emapplot-1.png" width="6000" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rm(GSEA_simplified, activated_terms, suppressed_terms, GSEA_activated, GSEA_suppressed, p1, p2)</code></pre>
<pre class="r"><code>GSEA_simplified &lt;- simplify(
  GSEA_all[[&quot;Macrophages&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)

GSEA_simplified &lt;- reorder_GO_by_pvalue(GSEA_simplified)

dotplot(GSEA_simplified, showCategory = 8, split=&quot;.sign&quot;, title = paste0(&quot;Macrophages&quot;, &quot;\nGO Enrichment Analysis&quot;, &quot;\nmPAP-associated genes&quot;), 
        label_format = 30, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;),
        strip.text = element_text(face = &quot;bold&quot;, size = 18)) +
  facet_grid(.~.sign)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/GSEA_Macrophages-1.png" width="2400" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GSEA_simplified &lt;- simplify(
  GSEA_all[[&quot;Lymphocytes&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)

GSEA_simplified &lt;- reorder_GO_by_pvalue(GSEA_simplified)

dotplot(GSEA_simplified, showCategory = 8, split=&quot;.sign&quot;, title = paste0(&quot;Lymphocytes&quot;, &quot;\nGO Enrichment Analysis&quot;, &quot;\nmPAP-associated genes&quot;), 
        label_format = 30, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;),
        strip.text = element_text(face = &quot;bold&quot;, size = 18)) +
  facet_grid(.~.sign)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/GSEA_Lymphocytes-1.png" width="2400" style="display: block; margin: auto;" /></p>
<pre class="r"><code>GSEA_simplified@result &lt;- classify_terms(GSEA_simplified@result)

activated_terms &lt;- subset(GSEA_simplified@result, sign == &quot;activated&quot;)

GSEA_activated &lt;- GSEA_simplified
GSEA_activated@result &lt;- activated_terms
GSEA_activated &lt;- pairwise_termsim(GSEA_activated)

p1 &lt;- emapplot(GSEA_activated, showCategory = 100, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Activated Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))

print(p1)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/GSEA_Lymphocytes_emapplot-1.png" width="3000" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rm(GSEA_simplified, activated_terms, suppressed_terms, GSEA_activated, GSEA_suppressed, p1, p2)</code></pre>
</div>
<div id="over-representation-analysis-unique-degs"
class="section level2">
<h2>Over-representation analysis unique DEGs</h2>
<pre class="r"><code>ORA_all_unique_up &lt;- list()
ORA_all_unique_down &lt;- list()
empty_enrich_result &lt;- new(
  &quot;enrichResult&quot;,
  result = data.frame(),  
  pvalueCutoff = 0.05,    
  pAdjustMethod = &quot;BH&quot;,   
  qvalueCutoff = 0.2,     
  gene = character(0),    
  organism = &quot;Homo sapiens&quot;,   
  keytype = &quot;ENTREZID&quot;,    
  geneSets = list(),      
  readable = TRUE
)

for (i in 1:length(cluster_names)) {
  universe_entrez &lt;- read.csv(here::here(output_dir_data, paste0(cluster_names[i], &quot;_universe_entrez.csv&quot;)))

  up_genes &lt;- signif[[i]] %&gt;%
    filter(logFC &gt; 0.5, unique_overlapping == &quot;Unique&quot;) %&gt;%
    pull(gene) %&gt;%  
    bitr(fromType = &quot;SYMBOL&quot;, toType = &quot;ENTREZID&quot;, OrgDb = &quot;org.Hs.eg.db&quot;) %&gt;%  
    pull(ENTREZID)
  down_genes &lt;- signif[[i]] %&gt;%
    filter(logFC &lt; 0.5, unique_overlapping == &quot;Unique&quot;) %&gt;%
    pull(gene) %&gt;%  
    bitr(fromType = &quot;SYMBOL&quot;, toType = &quot;ENTREZID&quot;, OrgDb = &quot;org.Hs.eg.db&quot;) %&gt;%  
    pull(ENTREZID)
  
  ORA_all_unique_up[[i]] &lt;- enrichGO(up_genes, OrgDb = &quot;org.Hs.eg.db&quot;, universe = universe_entrez, ont = &quot;ALL&quot;, readable = TRUE)
   if (length(ORA_all_unique_up) == i) {
     ORA_all_unique_up[[i]] &lt;- gsfilter(ORA_all_unique_up[[i]], by = &#39;Count&#39;, min = 3)
     ORA_all_unique_up[[i]] &lt;- reorder_GO_by_pvalue(ORA_all_unique_up[[i]])
   }
  if (length(ORA_all_unique_up) &lt; i) {
    ORA_all_unique_up[[i]] &lt;- empty_enrich_result
  }
  saveRDS(ORA_all_unique_up[[i]], here::here(output_dir_data, paste0(cluster_names[i], &quot;_ORA_Up_Unique_mPAP.rds&quot;)))
  names(ORA_all_unique_up)[i] &lt;- cluster_names[i]

  ORA_all_unique_down[[i]] &lt;- enrichGO(down_genes, OrgDb = &quot;org.Hs.eg.db&quot;, universe = universe_entrez, ont = &quot;ALL&quot;, readable = TRUE)
   if (length(ORA_all_unique_down) == i) {
     ORA_all_unique_down[[i]] &lt;- gsfilter(ORA_all_unique_down[[i]], by = &#39;Count&#39;, min = 3)
     ORA_all_unique_down[[i]] &lt;- reorder_GO_by_pvalue(ORA_all_unique_down[[i]])
   }
  if (length(ORA_all_unique_down) &lt; i) {
    ORA_all_unique_down[[i]] &lt;- empty_enrich_result
  }
  saveRDS(ORA_all_unique_down[[i]], here::here(output_dir_data, paste0(cluster_names[i], &quot;_ORA_Down_Unique_mPAP.rds&quot;)))
  names(ORA_all_unique_down)[i] &lt;- cluster_names[i]
}

rm(universe_entrez, up_genes, down_genes)

saveRDS(ORA_all_unique_up, here::here(output_dir_data, &quot;ORA_Up_Unique_mPAP_list.rds&quot;))
saveRDS(ORA_all_unique_down, here::here(output_dir_data, &quot;ORA_Down_Unique_mPAP_list.rds&quot;))</code></pre>
<pre class="r"><code>ORA_all_unique_up &lt;- readRDS(here::here(output_dir_data, &quot;ORA_Up_Unique_mPAP_list.rds&quot;))
ORA_all_unique_down &lt;- readRDS(here::here(output_dir_data, &quot;ORA_Down_Unique_mPAP_list.rds&quot;))</code></pre>
<pre class="r"><code>ORA_up_simplified &lt;- simplify(
  ORA_all_unique_up[[&quot;Cardiomyocytes&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)
ORA_up_simplified &lt;- reorder_GO_by_pvalue(ORA_up_simplified)

ORA_down_simplified &lt;- simplify(
  ORA_all_unique_down[[&quot;Cardiomyocytes&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)
ORA_down_simplified &lt;- reorder_GO_by_pvalue(ORA_down_simplified)

p1 &lt;- dotplot(ORA_up_simplified, showCategory = 10, title = paste0(&quot;Cardiomyocytes&quot;, &quot;\nOver-representation analysis&quot;, &quot;\nHigh mPAP - associated genes&quot;), label_format = 27, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;))
p2 &lt;- dotplot(ORA_down_simplified, showCategory = 10, title = paste0(&quot;Cardiomyocytes&quot;, &quot;\nOver-representation analysis&quot;, &quot;\nLow mPAP - associated genes&quot;), label_format = 27, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;))
print(p1|p2)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/ORA_Cardiomyocytes-1.png" width="3600" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ORA_up_simplified &lt;- pairwise_termsim(ORA_up_simplified)
ORA_down_simplified &lt;- pairwise_termsim(ORA_down_simplified)

p1 &lt;- emapplot(ORA_up_simplified, showCategory = 100, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Activated Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))
p2 &lt;- emapplot(ORA_down_simplified, showCategory = 100, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Suppressed Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))

print(p1 + p2)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/ORA_Cardiomyocytes_emapplot-1.png" width="6000" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rm(p1, p2)</code></pre>
<pre class="r"><code>ORA_up_simplified &lt;- simplify(
  ORA_all_unique_up[[&quot;Fibroblasts&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)
ORA_up_simplified &lt;- reorder_GO_by_pvalue(ORA_up_simplified)

ORA_down_simplified &lt;- simplify(
  ORA_all_unique_down[[&quot;Fibroblasts&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)
ORA_down_simplified &lt;- reorder_GO_by_pvalue(ORA_down_simplified)

p1 &lt;- dotplot(ORA_up_simplified, showCategory = 10, title = paste0(&quot;Fibroblasts&quot;, &quot;\nOver-representation analysis&quot;, &quot;\nHigh mPAP - associated genes&quot;), label_format = 27, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;))
p2 &lt;- dotplot(ORA_down_simplified, showCategory = 10, title = paste0(&quot;Fibroblasts&quot;, &quot;\nOver-representation analysis&quot;, &quot;\nLow mPAP - associated genes&quot;), label_format = 27, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;))
print(p1|p2)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/ORA_Fibroblasts-1.png" width="3600" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ORA_up_simplified &lt;- pairwise_termsim(ORA_up_simplified)
ORA_down_simplified &lt;- pairwise_termsim(ORA_down_simplified)

p1 &lt;- emapplot(ORA_up_simplified, showCategory = 100, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Activated Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))
p2 &lt;- emapplot(ORA_down_simplified, showCategory = 100, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Suppressed Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))

print(p1 + p2)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/ORA_Fibroblasts_emapplot-1.png" width="3600" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rm(p1, p2)</code></pre>
<pre class="r"><code>ORA_up_simplified &lt;- simplify(
  ORA_all_unique_up[[&quot;Endothelial_cells&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)
ORA_up_simplified &lt;- reorder_GO_by_pvalue(ORA_up_simplified)

ORA_down_simplified &lt;- simplify(
  ORA_all_unique_down[[&quot;Endothelial_cells&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)
ORA_down_simplified &lt;- reorder_GO_by_pvalue(ORA_down_simplified)

p1 &lt;- dotplot(ORA_up_simplified, showCategory = 10, title = paste0(&quot;Endothelial cells&quot;, &quot;\nOver-representation analysis&quot;, &quot;\nHigh mPAP - associated genes&quot;), label_format = 27, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;))
p2 &lt;- dotplot(ORA_down_simplified, showCategory = 10, title = paste0(&quot;Endothelial cells&quot;, &quot;\nOver-representation analysis&quot;, &quot;\nLow mPAP - associated genes&quot;), label_format = 27, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;))
print(p1|p2)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/ORA_Endothelial_cells-1.png" width="3600" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ORA_up_simplified &lt;- pairwise_termsim(ORA_up_simplified)
ORA_down_simplified &lt;- pairwise_termsim(ORA_down_simplified)

p1 &lt;- emapplot(ORA_up_simplified, showCategory = 50, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Activated Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))
p2 &lt;- emapplot(ORA_down_simplified, showCategory = 50, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Suppressed Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))

print(p1 + p2)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/ORA_Endothelial_emapplot-1.png" width="4800" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rm(p1, p2)</code></pre>
<pre class="r"><code>ORA_up_simplified &lt;- simplify(
  ORA_all_unique_up[[&quot;Pericytes&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)
ORA_up_simplified &lt;- reorder_GO_by_pvalue(ORA_up_simplified)

ORA_down_simplified &lt;- simplify(
  ORA_all_unique_down[[&quot;Pericytes&quot;]],
  cutoff = 0.75,
  by = &quot;p.adjust&quot;,
  select_fun = min,
  measure = &quot;Wang&quot;,
  semData = NULL
)
ORA_down_simplified &lt;- reorder_GO_by_pvalue(ORA_down_simplified)

p1 &lt;- dotplot(ORA_up_simplified, showCategory = 10, title = paste0(&quot;Pericytes&quot;, &quot;\nOver-representation analysis&quot;, &quot;\nHigh mPAP - associated genes&quot;), label_format = 27, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;))
p2 &lt;- dotplot(ORA_down_simplified, showCategory = 10, title = paste0(&quot;Pericytes&quot;, &quot;\nOver-representation analysis&quot;, &quot;\nLow mPAP - associated genes&quot;), label_format = 27, font.size = 15) +
  theme(plot.title = element_text( face = &quot;bold&quot;, size = 18, hjust = 0.5),
        axis.text.y = element_text(face = &quot;bold&quot;))
print(p1|p2)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/ORA_Pericytes-1.png" width="3600" style="display: block; margin: auto;" /></p>
<pre class="r"><code>ORA_up_simplified &lt;- pairwise_termsim(ORA_up_simplified)
ORA_down_simplified &lt;- pairwise_termsim(ORA_down_simplified)

p1 &lt;- emapplot(ORA_up_simplified, showCategory = 50, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Activated Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))
p2 &lt;- emapplot(ORA_down_simplified, showCategory = 50, cex_label_category = 0.7, min_edge = 0.15) +
  ggtitle(&quot;Suppressed Terms&quot;) +
  theme(plot.title = element_text(face = &quot;bold&quot;, size = 18, hjust = 0.5))

print(p1 + p2)</code></pre>
<p><img src="figure/DE_mPAP_all_cell_types.Rmd/ORA_Pericytes_emapplot-1.png" width="4800" style="display: block; margin: auto;" /></p>
<pre class="r"><code>rm(p1, p2)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span>
Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 4.3.1 (2023-06-16)
Platform: x86_64-pc-linux-gnu (64-bit)
Running under: Ubuntu 22.04.3 LTS

Matrix products: default
BLAS:   /usr/lib/x86_64-linux-gnu/openblas-pthread/libblas.so.3 
LAPACK: /usr/lib/x86_64-linux-gnu/openblas-pthread/libopenblasp-r0.3.20.so;  LAPACK version 3.10.0

locale:
 [1] LC_CTYPE=en_US.UTF-8          LC_NUMERIC=C                 
 [3] LC_TIME=en_US.UTF-8           LC_COLLATE=en_US.UTF-8       
 [5] LC_MONETARY=en_US.UTF-8       LC_MESSAGES=en_US.UTF-8      
 [7] LC_PAPER=en_US.UTF-8          LC_NAME=en_US.UTF-8          
 [9] LC_ADDRESS=en_US.UTF-8        LC_TELEPHONE=en_US.UTF-8     
[11] LC_MEASUREMENT=en_US.UTF-8    LC_IDENTIFICATION=en_US.UTF-8

time zone: Etc/UTC
tzcode source: system (glibc)

attached base packages:
[1] grid      stats4    stats     graphics  grDevices utils     datasets 
[8] methods   base     

other attached packages:
 [1] cowplot_1.1.3               rtracklayer_1.62.0         
 [3] ComplexHeatmap_2.18.0       UpSetR_1.4.0               
 [5] reactable_0.4.4             enrichplot_1.22.0          
 [7] dorothea_1.14.1             OmnipathR_3.10.1           
 [9] decoupleR_2.9.7             EnhancedVolcano_1.20.0     
[11] ggrepel_0.9.5               speckle_1.2.0              
[13] edgeR_4.0.16                limma_3.58.1               
[15] ReactomePA_1.46.0           AnnotationHub_3.10.0       
[17] BiocFileCache_2.10.1        dbplyr_2.4.0               
[19] org.Hs.eg.db_3.18.0         AnnotationDbi_1.64.1       
[21] clusterProfiler_4.10.1      gprofiler2_0.2.3           
[23] harmony_1.2.0               clustree_0.5.1             
[25] ggraph_2.2.1                rhdf5_2.46.1               
[27] Rcpp_1.0.12                 data.table_1.15.2          
[29] plyr_1.8.9                  gtable_0.3.4               
[31] gtools_3.9.5                ArchR_1.0.2                
[33] statmod_1.5.0               patchwork_1.2.0            
[35] scater_1.30.1               scuttle_1.12.0             
[37] Matrix.utils_0.9.7          RColorBrewer_1.1-3         
[39] scales_1.3.0                knitr_1.45                 
[41] gridExtra_2.3               png_0.1-8                  
[43] pheatmap_1.0.12             reshape2_1.4.4             
[45] lubridate_1.9.3             forcats_1.0.0              
[47] stringr_1.5.1               purrr_1.0.2                
[49] tidyr_1.3.1                 tibble_3.2.1               
[51] tidyverse_2.0.0             magrittr_2.0.3             
[53] ggplot2_3.5.0               dplyr_1.1.4                
[55] scCustomize_2.1.2           scDblFinder_1.16.0         
[57] Matrix_1.6-5                DropletUtils_1.22.0        
[59] SingleCellExperiment_1.24.0 SummarizedExperiment_1.32.0
[61] Biobase_2.62.0              GenomicRanges_1.54.1       
[63] GenomeInfoDb_1.38.7         IRanges_2.36.0             
[65] S4Vectors_0.40.2            BiocGenerics_0.48.1        
[67] MatrixGenerics_1.14.0       matrixStats_1.2.0          
[69] SeuratObject_5.0.2          Seurat_4.4.0               
[71] xlsx_0.6.5                  readxl_1.4.3               
[73] readr_2.1.5                 here_1.0.1                 

loaded via a namespace (and not attached):
  [1] igraph_2.0.3                  graph_1.80.0                 
  [3] ica_1.0-3                     plotly_4.10.4                
  [5] rematch2_2.1.2                zlibbioc_1.48.0              
  [7] tidyselect_1.2.1              rvest_1.0.4                  
  [9] bit_4.0.5                     doParallel_1.0.17            
 [11] clue_0.3-65                   lattice_0.22-5               
 [13] rjson_0.2.21                  blob_1.2.4                   
 [15] S4Arrays_1.2.1                parallel_4.3.1               
 [17] cli_3.6.2                     ggplotify_0.1.2              
 [19] goftest_1.2-3                 BiocIO_1.12.0                
 [21] bluster_1.12.0                grr_0.9.5                    
 [23] BiocNeighbors_1.20.2          uwot_0.1.16                  
 [25] shadowtext_0.1.3              curl_5.2.1                   
 [27] mime_0.12                     evaluate_0.23                
 [29] tidytree_0.4.6                leiden_0.4.3.1               
 [31] stringi_1.8.3                 backports_1.4.1              
 [33] XML_3.99-0.16.1               httpuv_1.6.14                
 [35] paletteer_1.6.0               rappdirs_0.3.3               
 [37] splines_4.3.1                 logger_0.3.0                 
 [39] bcellViper_1.38.0             sctransform_0.4.1            
 [41] ggbeeswarm_0.7.2              DBI_1.2.2                    
 [43] HDF5Array_1.30.1              jquerylib_0.1.4              
 [45] reactome.db_1.86.2            withr_3.0.0                  
 [47] git2r_0.33.0                  rprojroot_2.0.4              
 [49] xgboost_1.7.7.1               lmtest_0.9-40                
 [51] ggnewscale_0.4.10             tidygraph_1.3.1              
 [53] BiocManager_1.30.22           htmlwidgets_1.6.4            
 [55] fs_1.6.3                      labeling_0.4.3               
 [57] SparseArray_1.2.4             cellranger_1.1.0             
 [59] reticulate_1.35.0             zoo_1.8-12                   
 [61] XVector_0.42.0                timechange_0.3.0             
 [63] foreach_1.5.2                 fansi_1.0.6                  
 [65] ggtree_3.10.1                 R.oo_1.26.0                  
 [67] irlba_2.3.5.1                 ggrastr_1.0.2                
 [69] gridGraphics_0.5-1            ellipsis_0.3.2               
 [71] lazyeval_0.2.2                yaml_2.3.8                   
 [73] survival_3.5-8                scattermore_1.2              
 [75] BiocVersion_3.18.1            crayon_1.5.2                 
 [77] RcppAnnoy_0.0.22              progressr_0.14.0             
 [79] tweenr_2.0.3                  later_1.3.2                  
 [81] ggridges_0.5.6                codetools_0.2-19             
 [83] GlobalOptions_0.1.2           KEGGREST_1.42.0              
 [85] Rtsne_0.17                    shape_1.4.6.1                
 [87] Rsamtools_2.18.0              filelock_1.0.3               
 [89] pkgconfig_2.0.3               xml2_1.3.6                   
 [91] GenomicAlignments_1.38.2      aplot_0.2.2                  
 [93] spatstat.sparse_3.0-3         ape_5.7-1                    
 [95] viridisLite_0.4.2             xtable_1.8-4                 
 [97] highr_0.10                    httr_1.4.7                   
 [99] tools_4.3.1                   globals_0.16.3               
[101] beeswarm_0.4.0                checkmate_2.3.1              
[103] nlme_3.1-164                  HDO.db_0.99.1                
[105] digest_0.6.35                 farver_2.1.1                 
[107] tzdb_0.4.0                    yulab.utils_0.1.4            
[109] viridis_0.6.5                 glue_1.7.0                   
[111] cachem_1.0.8                  polyclip_1.10-6              
[113] generics_0.1.3                Biostrings_2.70.2            
[115] parallelly_1.37.1             ScaledMatrix_1.10.0          
[117] pbapply_1.7-2                 spam_2.10-0                  
[119] gson_0.1.0                    dqrng_0.3.2                  
[121] utf8_1.2.4                    graphlayouts_1.1.1           
[123] shiny_1.8.0                   GenomeInfoDbData_1.2.11      
[125] R.utils_2.12.3                rhdf5filters_1.14.1          
[127] RCurl_1.98-1.14               memoise_2.0.1                
[129] rmarkdown_2.26                R.methodsS3_1.8.2            
[131] future_1.33.1                 RANN_2.6.1                   
[133] Cairo_1.6-2                   spatstat.data_3.0-4          
[135] rstudioapi_0.15.0             cluster_2.1.6                
[137] whisker_0.4.1                 janitor_2.2.0                
[139] spatstat.utils_3.0-4          hms_1.1.3                    
[141] fitdistrplus_1.1-11           munsell_0.5.0                
[143] colorspace_2.1-0              rlang_1.1.4                  
[145] DelayedMatrixStats_1.24.0     sparseMatrixStats_1.14.0     
[147] dotCall64_1.1-1               ggforce_0.4.2                
[149] circlize_0.4.16               xfun_0.42                    
[151] iterators_1.0.14              abind_1.4-5                  
[153] GOSemSim_2.28.1               interactiveDisplayBase_1.40.0
[155] treeio_1.26.0                 rJava_1.0-11                 
[157] Rhdf5lib_1.24.2               bitops_1.0-7                 
[159] promises_1.2.1                scatterpie_0.2.1             
[161] RSQLite_2.3.5                 qvalue_2.34.0                
[163] fgsea_1.28.0                  DelayedArray_0.28.0          
[165] GO.db_3.18.0                  compiler_4.3.1               
[167] prettyunits_1.2.0             beachmat_2.18.1              
[169] graphite_1.48.0               listenv_0.9.1                
[171] workflowr_1.7.1               BiocSingular_1.18.0          
[173] tensor_1.5                    MASS_7.3-60.0.1              
[175] progress_1.2.3                BiocParallel_1.36.0          
[177] spatstat.random_3.2-3         R6_2.5.1                     
[179] fastmap_1.1.1                 fastmatch_1.1-4              
[181] vipor_0.4.7                   ROCR_1.0-11                  
[183] rsvd_1.0.5                    KernSmooth_2.23-22           
[185] miniUI_0.1.1.1                deldir_2.0-4                 
[187] htmltools_0.5.8.1             bit64_4.0.5                  
[189] spatstat.explore_3.2-6        lifecycle_1.0.4              
[191] ggprism_1.0.4                 restfulr_0.0.15              
[193] xlsxjars_0.6.1                sass_0.4.9                   
[195] vctrs_0.6.5                   spatstat.geom_3.2-9          
[197] snakecase_0.11.1              DOSE_3.28.2                  
[199] scran_1.30.2                  ggfun_0.1.4                  
[201] sp_2.1-3                      future.apply_1.11.1          
[203] bslib_0.6.1                   pillar_1.9.0                 
[205] metapod_1.10.1                locfit_1.5-9.9               
[207] jsonlite_1.8.8                GetoptLong_1.0.5             </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
https://docs.mathjax.org/en/latest/web/configuration.html. This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>




</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // temporarily add toc-ignore selector to headers for the consistency with Pandoc
    $('.unlisted.unnumbered').addClass('toc-ignore')

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_');
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
